Sources/AsyncHTTPClient/SSLContextCache.swift

```diff
@@ -0,0 +1,102 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Logging
+import NIO
+import NIOConcurrencyHelpers
+import NIOSSL
+
+class SSLContextCache {
+    private var state = State.activeNoThread
+    private let lock = Lock()
+    private var sslContextCache = LRUCache<BestEffortHashableTLSConfiguration, NIOSSLContext>()
+    private let threadPool = NIOThreadPool(numberOfThreads: 1)
+
+    enum State {
+        case activeNoThread
+        case active
+        case shutDown
```

Comment:
Just out of interest, when do we use `shutdown` and when `shutDown`?

Reply:
The problem is that in English, it can be "shutdown" (noun), "shut down", and "shut-down", eg. "Let's initiate the shutdown.", "This is a shut-down factory.", or "The factory is shut down.". I was aiming for "The cache is shut down" and because it's two words, I made it `.shutDown` :). But honestly, this totally doesn't matter and we can also make it `.shutdown` or `.closed`, I don't mind either way.